demo/client/src/view.ts

```diff
@@ -197,6 +202,60 @@ export abstract class DemoView extends PathfinderView implements RenderContext {
 
         this.atlasRenderingTimerQuery = null;
         this.compositingTimerQuery = null;
+
+        this.inVRRAF = false;
+        this.inVR = false;
+        this.vrDisplay = null;
+        if ("VRFrameData" in window) {
+           this.vrFrameData = new VRFrameData;
+        } else {
+            this.vrFrameData = null;
+        }
+
+        this.vrSetup();
+    }
+
+    vrSetup(): void {
+        if (navigator.getVRDisplays) {
+            navigator.getVRDisplays().then((displays) => {
+              if (displays.length > 0) {
+                this.vrDisplay = displays[displays.length - 1];
+
+                // It's heighly reccommended that you set the near and far planes to
+                // something appropriate for your scene so the projection matricies
+                // WebVR produces have a well scaled depth buffer.
+                this.vrDisplay.depthNear = NEAR_CLIP_PLANE;
+                this.vrDisplay.depthFar = FAR_CLIP_PLANE;
+                unwrapNull(document.getElementById('pf-vr')).style.display = "initial";
+              } else {
+                // no vr displays
+              }
+            });
+        } else {
+            // no vr support
+        }
+
+        window.addEventListener('vrdisplaypresentchange', () => {
+
+          if (this.vrDisplay == null)
+              return;
+
+          if (this.vrDisplay.isPresenting) {
+            const that = this;
+            function vrCallback(): void {
```

Comment:
Can't we use an arrow function here to get lexical `this`? `const vrCallback = () => { ... }`